client/extensions/woocommerce/state/sites/settings/mailchimp/test/actions.js

```diff
@@ -347,8 +347,85 @@ describe( 'actions', () => {
 	describe( '#mailChimpSaveSettings()', () => {
 		const siteId = '123';
 
-		test( 'should dispatch an action', () => {
+		test( 'should not dispatch an action if ther are not settings in state', () => {
 			const getState = () => ( {} );
+			const dispatch = spy();
+			mailChimpSaveSettings( siteId )( dispatch, getState );
+			expect( dispatch ).to.have.not.been.calledWith( {
+				type: WOOCOMMERCE_MAILCHIMP_SAVE_SETTINGS,
+				siteId,
+			} );
+		} );
+
+		test( 'should not dispatch an action if the state does not represent compleated setup', () => {
+			const getState = () => ( {
+				extensions: {
+					woocommerce: {
+						sites: {
+							[ siteId ]: {
+								settings: {
+									mailchimp: {
+										settings: {
+											mailchimp_api_key: '12345testing',
+											mailchimp_debugging: false,
+											mailchimp_account_info_id: '64c533c4',
+											mailchimp_account_info_username: 'ski',
+											mailchimp_active_tab: 'campaign_defaults',
+											store_name: 'mystore',
+											store_street: 'street',
+											store_city: 'Valhalla',
+											campaign_from_name: 'woo',
+											campaign_from_email: 'mystore@woo.woo',
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			} );
+
+			const dispatch = spy();
+			mailChimpSaveSettings( siteId )( dispatch, getState );
+			expect( dispatch ).to.have.not.been.calledWith( {
+				type: WOOCOMMERCE_MAILCHIMP_SAVE_SETTINGS,
+				siteId,
+			} );
+		} );
+
+		test( 'should dispatch an action if the state represents compleated setup', () => {
+			const getState = () => ( {
+				extensions: {
+					woocommerce: {
+						sites: {
+							[ siteId ]: {
+								settings: {
+									mailchimp: {
+										settings: {
+											mailchimp_api_key: '6e46d0621d-us16',
+											mailchimp_debugging: false,
+											mailchimp_account_info_id: '64c533c4',
+											mailchimp_account_info_username: 'ski',
+											active_tab: 'sync',
+											store_name: 'sdsd',
+											campaign_from_name: 'sdsd',
+											campaign_from_email: 'user@gmail.com',
+											campaign_subject: 'sdsd',
+											campaign_language: 'en',
+											campaign_permission_reminder:
+												'You were subscribed to the newsletter from sdsd.',
+											mailchimp_list: '633bwew8',
+											newsletter_label: 'Subscribe to our newsletter',
+											mailchimp_active_tab: 'sync',
```

Comment:
Same comment as above

Reply:
In this object you have both `active_tab` and `mailchimp_active_tab` so testing here against `active_tab` is OK. Presence of `active_tab` in state `sync` indicates that plugin is configured. Having both fields is necessary as one indicates "local" ( Calypso ) state and the other one indicates state on site. It is confusing I know - I could make it more explicit with code in wc-calypso-bridge but I did not wanted to add another layer of code on top of what MailChimp plugin is making. Overall things like this happen because plugin was not originally designed to be controlled with REST endpoints.